modules/Channel.md

```diff
@@ -0,0 +1,112 @@
+## Channel Module
+
+A channel is a FIFO data structure used to communicate between different Chapel
+tasks. You can learn more about how go-channels work by following this
+[link](https://gobyexample.com/channels)
```

Comment:
A classic question here:  Are we likely to end up similar enough to Go's channels that it makes sense to lean on Go's documentation outright here, rather than describing Chapel's channels outright?  (in which case we might link to the Go channels as a strong source of inspiration rather than suggesting that the best way to learn about Chapel channels is to go read Go's documentation)?

Also, there's a missing period at the end of this hyperlink (which seems more problematic in the rendered documents than in these ones.